File: ios/RBDarts/Features/Corrections/CorrectionDiagnostics.swift
import Foundation

public struct CorrectionDiagnostics {
    private let reporter: DiagnosticsReporting

    public init(reporter: DiagnosticsReporting = PrivacySafeDiagnostics()) {
        self.reporter = reporter
    }

    public func recordRequested(gameId: String) {
        reporter.record(DiagnosticEvent(name: "correction_requested", severity: .info, attributes: ["gameId": gameId]))
    }

    public func recordDenied(gameId: String) {
        reporter.record(DiagnosticEvent(name: "correction_denied", severity: .warning, attributes: ["gameId": gameId]))
    }

    public func recordRecalculationFailure(gameId: String) {
        reporter.record(DiagnosticEvent(name: "correction_recalculation_failed", severity: .error, attributes: ["gameId": gameId]))
    }
}

File: android/app/src/main/java/com/rbdarts/feature/corrections/CorrectionDiagnostics.kt
package com.rbdarts.feature.corrections

import com.rbdarts.core.observability.DiagnosticEvent
import com.rbdarts.core.observability.DiagnosticSeverity
import com.rbdarts.core.observability.DiagnosticsReporter
import com.rbdarts.core.observability.PrivacySafeDiagnostics

class CorrectionDiagnostics(private val reporter: DiagnosticsReporter = PrivacySafeDiagnostics()) {
    fun recordRequested(gameId: String) {
        reporter.record(DiagnosticEvent("correction_requested", DiagnosticSeverity.Info, mapOf("gameId" to gameId)))
    }

    fun recordDenied(gameId: String) {
        reporter.record(DiagnosticEvent("correction_denied", DiagnosticSeverity.Warning, mapOf("gameId" to gameId)))
    }

    fun recordRecalculationFailure(gameId: String) {
        reporter.record(DiagnosticEvent("correction_recalculation_failed", DiagnosticSeverity.Error, mapOf("gameId" to gameId)))
    }
}
